src/main/java/org/openmrs/maven/plugins/SetupPlatform.java

```diff
@@ -0,0 +1,260 @@
+     * @param serverId
+     * @param version
+     * @param dbDriver
+     * @param dbUri
+     * @param dbUser
+     * @param dbPassword
+     * @param interactiveMode
+     * @throws MojoExecutionException
+     */
+    public Server setup(String serverId,
+                      String version,
+                      String dbDriver,
+                      String dbUri,
+                      String dbUser,
+                      String dbPassword,
+                      String interactiveMode) throws MojoExecutionException {
+        // path to omrs home
+        File omrsPath = new File(System.getProperty("user.home"), SDKConstants.OPENMRS_SERVER_PATH);
+        // check if user not set serverId parameter
+        if (serverId == null) try {
+            // default server name
+            String defaultId = "server";
+            // index for search unused server name
+            int indx = 0;
+            // find unused name for server
+            while (new File(omrsPath, defaultId).exists()) {
+                indx++;
+                defaultId = "server" + String.valueOf(indx);
+            }
+            // prompt this param
+            serverId = prompter.prompt("Define value for property 'serverId': (default: '" + defaultId + "')");
+            // check if user pressed 'enter' (default value)
+            if (serverId.equals("")) serverId = defaultId;
+        } catch (PrompterException e) {
+            e.printStackTrace();
+        }
+        // path to server with serverId
+        File serverPath = new File(omrsPath, serverId);
+        // check existence
+        if (serverPath.exists()) throw new MojoExecutionException("Server with same id already created");
+        // execute plugin for create server
+        executeMojo(
+                plugin(
+                        groupId(SDKConstants.ARCH_GROUP_ID),
+                        artifactId(SDKConstants.ARCH_ARTIFACT_ID),
+                        version(SDKConstants.ARCH_VERSION)
+                ),
+                goal("generate"),
+                configuration(
+                        element(name("archetypeCatalog"), SDKConstants.ARCH_CATALOG),
+                        element(name("interactiveMode"), interactiveMode),
+                        //element(name("package"), SDKConstants.PROJECT_PACKAGE),
+                        element(name("archetypeGroupId"), SDKConstants.ARCH_PROJECT_GROUP_ID),
+                        element(name("archetypeArtifactId"), SDKConstants.ARCH_PROJECT_ARTIFACT_ID),
+                        element(name("archetypeVersion"), SDKConstants.ARCH_PROJECT_VERSION)
+                        //element(name("groupId"), SDKConstants.PROJECT_GROUP_ID),
+                        //element(name("artifactId"), serverId),
+                        //element(name("version"), version)
+                ),
+                executionEnvironment(mavenProject, mavenSession, pluginManager)
+        );
+        // move server after creating
```

Comment:
Instead of writing an inline comment you should create a method called moveServer and call it here. It would be easier to read the setup method, which has too many lines of code in it.